### PR TITLE
Change default value of `encodeDefaults` config flag

### PR DESCRIPTION
**Pull request by sandwwraith**

### CODE REVIEW

**Review by qwwdfsad — CHANGES_REQUESTED** [review]
Please fix our guide tests

File: docs/basic-serialization.md [hunk 1]
```diff
@@ -437,10 +437,10 @@ fun main() {
 
 > You can get the full code [here](../guide/example/example-classes-09.kt).
 
-It produces the following output, which has `language` property, even though its value is equal to the default one.
+It produces the following output, which does not have `language` property, because its value is equal to the default one.
```
Comment (qwwdfsad) [inline docs/basic-serialization.md:440]: Please add a short sentence explaining the rationale of this behaviour

**Review by qwwdfsad — CHANGES_REQUESTED** [review]
Our arguments regarding encode defaults are also applicable to Protobuf, Cbor, Hocon etc. Please change it everywhere

File: docs/basic-serialization.md (hunk 1 above)
Comment (qwwdfsad) [reply docs/basic-serialization.md:440]: Friendly reminder

File: formats/cbor/commonMain/src/kotlinx/serialization/cbor/Cbor.kt [hunk 2]
```diff
@@ -25,6 +25,7 @@ import kotlinx.serialization.modules.*
  * from corresponding Kotlin objects. However, other 3rd-party parsers (e.g. `jackson-dataformat-cbor`) may not accept such maps.
  *
  * @param encodeDefaults specifies whether default values of Kotlin properties are encoded.
+ *                       False by default; meaning that properties with values equal to defaults would be elided.
```
Comment (qwwdfsad) [inline formats/cbor/commonMain/src/kotlinx/serialization/cbor/Cbor.kt:28]: would be -> will be

**Review by qwwdfsad — APPROVED** [review]
Otherwise looks good

File: docs/basic-serialization.md (hunk 1 above)
Comment (sandwwraith) [reply docs/basic-serialization.md:440]: Yeah thanks

**Comment by v4dkou** [comment]
> It produces the following output, which does not have `language` property, because its value is equal to the default one.

@sandwwraith Why is that a problem that the value is equal to the default one?

Almost submitted this as a bug, because this behaviour messes things up when transferring objects from Kotlin to JS context as plain JSONs. An object has a value in Kotlin, but then it loses it in JS (unless the flag is specifically set to true), which is **really** unexpected.

**Pull request merged**